Jenkinsfile

```diff
@@ -2,159 +2,56 @@ import org.kohsuke.github.GitHub
-  !IS_DEV_BRANCH &&
-  !IS_STAGING_BRANCH &&
-  !IS_PROD_BRANCH &&
-  !env.CHANGE_TARGET &&
-  currentBuild.nextBuild
-}
-
-def runDeploy(jobName, ref) {
-  build job: jobName, parameters: [
-    booleanParam(name: 'notify_slack', value: true),
-    stringParam(name: 'ref', value: ref),
-  ], wait: false
-}
-
-def buildDetails = { vars ->
-  """
-    BUILDTYPE=${vars['buildtype']}
-    NODE_ENV=production
-    BRANCH_NAME=${env.BRANCH_NAME}
-    CHANGE_TARGET=${env.CHANGE_TARGET}
-    BUILD_ID=${env.BUILD_ID}
-    BUILD_NUMBER=${env.BUILD_NUMBER}
-    REF=${vars['ref']}
-  """.stripIndent()
-}
-
-def notify = { ->
-  if (IS_DEV_BRANCH || IS_STAGING_BRANCH || IS_PROD_BRANCH) {
-    message = "vets-website ${env.BRANCH_NAME} branch CI failed. |${env.RUN_DISPLAY_URL}".stripMargin()
-    slackSend message: message,
-    color: 'danger',
-    failOnError: true
-  }
-}
 
 node('vetsgov-general-purpose') {
   properties([[$class: 'BuildDiscarderProperty', strategy: [$class: 'LogRotator', daysToKeepStr: '60']],
               // a string param cannot be null, so we set the arbitrary value of 'none' here to make sure the default doesn't match anything
               [$class: 'ParametersDefinitionProperty', parameterDefinitions: [[$class: 'StringParameterDefinition', name: 'cmsEnv', defaultValue: 'none']]]]);
-  def dockerImage, args, ref, imageTag
-
-  def cmsEnv = params.get('cmsEnv', 'none')
-
-  // Checkout source, create output directories, build container
-
-  stage('Setup') {
-    try {
-      buildTypeOverride = DRUPAL_MAPPING.get(params.get('cmsEnv', 'none'), null)
-      if(buildTypeOverride) {
-        VAGOV_BUILDTYPES = [buildTypeOverride]
-      }
-
-      // Jenkins doesn't like it when we checkout the secondary repository first
-      // so we checkout 'vets-website' first
-      dir("vets-website") {
-        checkout scm
-      }
-
-      // clone vagov-content
-      checkout changelog: false, poll: false, scm: [$class: 'GitSCM', branches: [[name: '*/master']], doGenerateSubmoduleConfigurations: false, extensions: [[$class: 'CloneOption', noTags: true, reference: '', shallow: true], [$class: 'RelativeTargetDirectory', relativeTargetDir: 'vagov-content']], submoduleCfg: [], userRemoteConfigs: [[url: 'git@github.com:department-of-veterans-affairs/vagov-content.git']]]
 
-      args = "-v ${pwd()}/vets-website:/application -v ${pwd()}/vagov-content:/vagov-content"
+  def ref
 
-      dir("vets-website") {
-        ref = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+  dir("vets-website") {
+    checkout scm
+    ref = sh(returnStdout: true, script: 'git rev-parse HEAD').trim()
+  }
 
-        sh "mkdir -p build"
-        sh "mkdir -p logs/selenium"
-        sh "mkdir -p coverage"
-        sh "mkdir -p temp"
+  def buildUtil = load "vets-website/Jenkinsfile.common"
+  def dockerArgs = "-v ${WORKSPACE}/vets-website:/application -v ${WORKSPACE}/vagov-content:/vagov-content"
```

Comment:
Since these have to be passed to various stages, it may make sense to move this logic to set the string in the common file, that way the args definition isn't duplicated.

Reply:
Yeah the pwd() was during the wild goose chase trying to fix docker. Good call.

Reply:
@wyattwalter should I just get rid of most if not all of the parameters for these functions that run stages? Since most of the needed info is in the common.groovy file anyways.

Reply:
I think that would be ideal. I know that the main Jenkinsfile does need at least one of those parameters, though :(